blog/03-26-2023-Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2/Build-A-Decentralized-Freelancer-Marketplace-On-The-Celo-Blockchain-Part-2.md

```diff
@@ -0,0 +1,797 @@
+    }
+  };
+```
+
+We define an async function called `addFreelancer` that takes the freelancer's details as arguments and calls the `newFreelancer` function from the smart contract.
+
+Changing a freelancer's description and price:
+
+```javascript
+onst changeDescription = async (_index, _description) => {
+    try {
+      await contract.methods
+        .changeDescription(_index, _description)
+        .send({ from: address });
+      getFreelancers();
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+
+```javascript
+const changePrice = async (_index, _newprice) => {
+    try {
+      await contract.methods
+        .changePrice(_index, _newprice)
+        .send({ from: address });
+      getFreelancers();
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+
+We define async functions `changeDescription` and `changePrice` that allow freelancers to update their description and hourly price.
+
+Toggling a freelancer's availability:
+
+```javascript
+const toggleAvailable = async (_index) => {
+    try {
+      await contract.methods
+        .toggleAvailable(_index)
+        .send({ from: address });
+      getFreelancers();
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+
+We define an async function called `toggleAvailable` that allows a freelancer to toggle their availability status.
+
+Hiring a freelancer:
+
+```javascript
+const hireFreelancer = async (_index, _hours) => {
+    try {
+      const cUSDContract = new kit.web3.eth.Contract(IERC, cUSDContractAddress);
+      const totalPayment = _hours * freelancers[_index].price;
+      const cost = new BigNumber(totalPayment)
+              .shiftedBy(ERC20_DECIMALS)
+              .toString();
+      console.log(totalPayment);
+      await cUSDContract.methods
+        .approve(contractAddress, cost)
+        .send({ from: address });
+      await contract.methods
+        .hireFreelancerHourly(_index, _hours, cost)
+        .send({ from: address });
+      getFreelancers();
+      getBalance();
+      alert("you have successfully sent cusd to this user");
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+
+We define an async function called `hireFreelancer` that calculates the total payment based on the freelancer's hourly rate and the number of hours worked, approves the payment in cUSD, and calls the `hireFreelancerHourly` function from the smart contract.
+
+#### useEffect hooks:
+
+```javascript
+ useEffect(() => {
+    connectToWallet();
+  }, []);
+
+  useEffect(() => {
+    if (kit && address) {
+      getBalance();
+    }
+  }, [kit, address, getBalance]);
+
+  useEffect(() => {
+    if (contract) {
+      getFreelancers();
+    }
+  }, [contract, getFreelancers]);
+```
+
+We use three `useEffect` hooks to automatically connect to the wallet, fetch the user's cUSD balance, and fetch the list of freelancers when the component is mounted or when any of the relevant state variables change.
+
+#### Rendering the components:
```

Comment:
can you make this a double hashtag or triple at most or maybe just use a bold text